src/components/popover/popover.jsx

```diff
@@ -66,24 +56,24 @@ const Popover = forwardRef(function Popover(props, ref) {
     <section
       aria-labelledby="popover-heading"
       className={styles.container}
+      onBlur={onBlur}
       ref={sectionRef}
       role="dialog"
       style={flush}
+      tabIndex={-1}
```

Comment:
This is really confusing me.

`onblur` is something that triggers when focus is lost, but the tabindex of -1 indicates to me it _can't_ be focused, so how dose `onblur` ever get triggered? I feel like I'm getting something mixed up here

Moreover, aren't sections out of the tabbing order already?

Reply:
A tabindex of -1 implies it can be focused with JavaScript and it _not_ part of the page tab sequence. It is needed to populate the `FocusEvent#relatedTarget` property too. 

Without this bit, clicking _in_ the Popover but something non-interactive implies that _nothing_ is focused. The activeElement is the document but the event does not have a relatedTarget (`null`). 

By setting the tab index, the browser thinks the Popover itself gets focused (and is actually the case in Safari). This means that there is a relatedTarget. In addition, once the user tries to focus anything else the `onBlur` is still invoked. 

It's kinda nice because it solves the "did the composite widget lose focus" question really easily. While focus keeps shifting within the component, we know we're focused. Otherwise, we know we lost focus -- regardless if the relatedTarget is not populated.